oauthbearer.go

```diff
@@ -0,0 +1,46 @@
+package sasl
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+// The OAUTHBEARER mechanism name.
+const OAuthBearer = "OAUTHBEARER"
+
+type OAuthBearerError struct {
+	Status  string `json:"status"`
+	Schemes string `json:"schemes"`
+	Scope   string `json:"scope"`
+}
+
+// Implements error
+func (err *OAuthBearerError) Error() string {
+	return fmt.Sprintf("OAUTHBEARER authentication error (%v)", err.Status)
+}
+
+type oauthbearerClient struct {
+	Username string
+	Token    string
+}
+
+func (a *oauthbearerClient) Start() (mech string, ir []byte, err error) {
+	mech = OAuthBearer
+	ir = []byte("n,a=" + a.Username + ",\x01auth=Bearer " + a.Token + "\x01\x01")
+	return
+}
+
+func (a *oauthbearerClient) Next(challenge []byte) ([]byte, error) {
+	authBearerErr := &OAuthBearerError{}
+	if err := json.Unmarshal(challenge, authBearerErr); err != nil {
+		return nil, err
+	} else {
+		return nil, authBearerErr
+	}
+}
+
+// An implementation of the OAUTHBEARER authentication mechanism, as
+// described in https://tools.ietf.org/html/rfc7628
+func NewOAuthBearerClient(username, token string) Client {
```

Comment:
This is missing host and port. Maybe a struct should be used instead of separate parameters?